include/deal.II/algorithms/general_data_storage.h

```diff
@@ -272,7 +272,7 @@ class GeneralDataStorage : public Subscriptor
   template <typename Type, typename Arg, typename... Args>
   Type &
   get_or_add_object_with_name(const std::string &name,
-                              Arg &&             argument,
+                              Arg DEAL_II_AND argument,
```

Comment:
This is just not a comparison and for me also the rest is so much less readable.